lib/semantics/tools.cc

```diff
@@ -508,6 +509,169 @@ std::optional<parser::MessageFixedText> WhyNotModifiable(
   }
 }
 
+bool IsCoarray(const parser::AllocateObject &allocateObject) {
```

Comment:
`IsCoarray` and some of the `IsImageControlStmt` overloads below should be declared static so that it is clear they are just helpers and not part of the public API.

Reply:
Thanks, Tim.  I'll change them.